test/spec/dataset/merge.js

```diff
@@ -411,6 +412,81 @@ module.exports = {
             assert(merge.itemCount == 0);
             assert(merge.sources.length == 0);
           }
+        },
+        {
+          name: 'setAccumulateState',
+          test: [
+            {
+              name: 'create merge in accumulate state',
+              test: function(){
+                function obj(text) {
+                  return new DataObject({ data: { text: text } });
+                }
+                var foo = obj('foo');
+                var bar = obj('bar');
+
+                var fooDataset = new Dataset({
+                  items: [
+                    foo
+                  ]
+                });
+                var barDataset = new Dataset({
+                  items: [
+                    bar
+                  ]
+                });
+
+                Dataset.setAccumulateState(true);
+                var merge = new Merge({
+                  sources: [
+                    fooDataset,
+                    barDataset
+                  ]
+                });
+                Dataset.setAccumulateState(false);
+
+                assert(merge.itemCount === 2);
+                assert(merge.has(foo) === true);
+                assert(merge.has(bar) === true);
+              },
+            },
+            {
+              name: 'create merge and delete items in accumulate state',
+              test: function(){
+                function obj(text) {
+                  return new DataObject({ data: { text: text } });
+                }
+                var foo = obj('foo');
+                var bar = obj('bar');
+
+                var fooDataset = new Dataset({
+                  items: [
+                    foo
+                  ]
+                });
+                var barDataset = new Dataset({
+                  items: [
+                    bar
+                  ]
+                });
+
+                Dataset.setAccumulateState(true);
+                var merge = new Merge({
+                  sources: [
+                    fooDataset,
+                    barDataset
+                  ]
+                });
+                debugger;
+                merge.setRule(Merge.INTERSECTION);
+                Dataset.setAccumulateState(false);
+
+                assert(merge.itemCount === 0);
```

Comment:
@smelukov Almost everywhere in tests we use `assert(<something> === true)` instead of `assert(<something)`